rust/lance/src/dataset/transaction.rs

```diff
@@ -551,7 +551,7 @@ impl Transaction {
         manifest.tag.clone_from(&self.tag);
 
         if config.auto_set_feature_flags {
-            apply_feature_flags(&mut manifest)?;
+            apply_feature_flags(&mut manifest, config.use_move_stable_row_ids)?;
```

Comment:
So what are the rules on this config option?  Is it the same as the legacy format rules (only applies to new datasets) or can you adjust the row id style on demand?

Reply:
Right now, it should only be sett-able on new datasets. And once set it cannot be unset.